base/loading.jl

```diff
@@ -742,7 +743,17 @@ function stale_cachefile(modpath::String, cachefile::String)
             DEBUG_LOADING[] && info("JL_DEBUG_LOADING: Rejecting cache file $cachefile due to it containing an invalid cache header.")
             return true # invalid cache file
         end
-        modules, files = parse_cache_header(io)
+        modules, files, required_modules = parse_cache_header(io)
+
+        # Check if transitive dependencies can be fullfilled
+        for mod in keys(required_modules)
+            isdefined(Main, mod) && continue
```

Comment:
I am checking if the module is already loaded here, since if it is `read_verify_mod_list` from `dump.c`, won't fail. Just added a commit to be more explicit (it safes a few calls to `find_in_node_path`)